File: Practice_team06/Models/Movie.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using NpgsqlTypes;

namespace Practice_team06.Models;

public partial class Movie
{
    public int Id { get; set; }
    [Required(ErrorMessage = "Назва фільму обов'язкова")]
    [StringLength(255)]
    public string Title { get; set; } = null!;

    public string? Description { get; set; }
    [Range(1, 600, ErrorMessage = "Тривалість від 1 до 600 хв")]
    public int? DurationMin { get; set; }
    [DataType(DataType.Date)]
    public DateOnly? ReleaseDate { get; set; }
    [Required]
    [Range(0, 10000)]
    public decimal BasePrice { get; set; }
    
    public DateOnly? StartDate { get; set; }

    public DateOnly? EndDate { get; set; }
    [Range(0, 10)]
    public decimal? Rating { get; set; }
    [Url(ErrorMessage = "Некоректне посилання")]
    public string? PosterUri { get; set; }
    [Url(ErrorMessage = "Некоректне посилання")]
    public string? TrailerUri { get; set; }
    
    public AgeRestriction AgeRestriction { get; set; }
    
    public int? DirectorId { get; set; }
    
    public virtual ICollection<MovieActor> MovieActors { get; set; } = new List<MovieActor>();

    public virtual ICollection<Session> Sessions { get; set; } = new List<Session>();

    public virtual ICollection<Genre> Genres { get; set; } = new List<Genre>();
    
    public virtual Director? Director { get; set; }  
}
public enum AgeRestriction { 
    [PgName("0+")] ZeroPlus, 
    [PgName("12+")] TwelvePlus, 
    [PgName("16+")] SixteenPlus, 
    [PgName("18+")] EighteenPlus 
}

File: cinema-frontend/src/types/movie.ts
import type {DirectorDto} from "./director.ts";
import type {ActorDto} from "./actor.ts";
import type {GenreDto} from "./genre.ts";
import type {BaseFilterDto} from "./common.ts";

export interface MovieActorDto {
    actorId: number;
    roleName: string;
    actor?: ActorDto;
}

export interface MovieDto {
    id: number;
    title: string;
    description?: string;
    durationMin: number;
    releaseDate?: string;
    basePrice: number;
    rating?: number;
    trailerUri?: string;
    posterUri?: string;
    ageRestriction: string;
    startDate?: string;
    endDate?: string;
    director?: DirectorDto;
    movieActors: MovieActorDto[];
    genres: GenreDto[]
    languageIds: number[];
    actors: ActorDto[];
}

export interface CreateMovieDto {
    title: string;
    description: string;
    durationMin: number;
    releaseDate: string;
    basePrice: number;
    startDate?: string;
    endDate?: string;
    posterUri?: string;
    trailerUri?: string;
    languageIds: number[];
    ageRestriction: number;
    directorId?: number;
    genreIds: number[];
    movieActors: { actorId: number; roleName: string }[];
    rating?: number;
}

export const AgeRestriction = {
    ZeroPlus: 0,
    TwelvePlus: 1,
    SixteenPlus: 2,
    EighteenPlus: 3
} as const;

export type AgeRestrictionType = typeof AgeRestriction[keyof typeof AgeRestriction];

export interface MovieFilterDto extends BaseFilterDto {
    title?: string;
    rating?: number;
    ageRestrictions?: AgeRestrictionType[];
    genres?: string[];
}